sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -53,7 +53,11 @@ class PathFilterIgnoreNonData(stagingDir: String) extends PathFilter with Serial
 object CommandUtils extends Logging {
 
   /** Change statistics after changing data by commands. */
-  def updateTableStats(sparkSession: SparkSession, table: CatalogTable): Unit = {
+  def updateTableStats(
```

Comment:
seems we also need to update a few other commands:
- `AlterTableSetLocationCommand`
- `LoadDataCommand`
- `TruncateTableCommand`
- `InsertIntoHadoopFsRelationCommand`

It'd be better to only update stats for those partitions that are being touched, instead of updating all partitions.

Reply:
Will look into it

Reply:
Updated and added tests